### PR TITLE
Add shield in composer

### LINKED ISSUE #1077: Composer decoration: shields
image

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #1077

image

image

image

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/composer/TextComposerView.kt [hunk 1]
```diff
@@ -158,12 +163,23 @@ class TextComposerView @JvmOverloads constructor(context: Context, attrs: Attrib
...
+            composerShieldImageView.isVisible = true
+            val shield = when (roomEncryptionTrustLevel) {
+                RoomEncryptionTrustLevel.Trusted -> {
+                    ContextCompat.getDrawable(context, R.drawable.ic_shield_trusted)
+                }
+                RoomEncryptionTrustLevel.Warning -> {
+                    ContextCompat.getDrawable(context, R.drawable.ic_shield_warning)
+                }
+                else -> ContextCompat.getDrawable(context, R.drawable.ic_shield_black)
+            }
```
Comment (Dominaezzz) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/composer/TextComposerView.kt:178]: A bit of factorisation.
```suggestion
            val shieldResId = when (roomEncryptionTrustLevel) {
                RoomEncryptionTrustLevel.Trusted -> R.drawable.ic_shield_trusted
                RoomEncryptionTrustLevel.Warning -> R.drawable.ic_shield_warning
                else -> R.drawable.ic_shield_black
            }
            val shield = ContextCompat.getDrawable(context, shieldResId)
```

Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/composer/TextComposerView.kt]: updated

File: vector/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -515,8 +515,8 @@
     <string name="room_one_user_is_typing">%s is typing…</string>
     <string name="room_two_users_are_typing">%1$s &#038; %2$s are typing…</string>
     <string name="room_many_users_are_typing">%1$s &#038; %2$s &#038; others are typing…</string>
-    <string name="room_message_placeholder_encrypted">Send an encrypted message…</string>
-    <string name="room_message_placeholder_not_encrypted">Send a message (unencrypted)…</string>
+    <string name="room_message_placeholder_encrypted">Message…</string>
+    <string name="room_message_placeholder_not_encrypted">Message…</string>
```
Comment (bmarty) [inline vector/src/main/res/values/strings.xml:519]: Please create new string resource(s?) for riotx

Comment (BillCarsonFr) [reply vector/src/main/res/values/strings.xml]: added new resource

File: matrix-sdk-android/src/main/res/values/strings_RiotX.xml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-
+    <string name="room_message_placeholder">Message…</string>"
```
Comment (bmarty) [inline matrix-sdk-android/src/main/res/values/strings_RiotX.xml:3]: Wrong file :/

Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/res/values/strings_RiotX.xml]: ah sry :/

**Pull request merged**